### ISSUE TITLE
File transfer hangs indefinitely when Marlin is configured with "SDCARD_CONNECTION LCD"

### ISSUE THREAD

**Issue by jpflouret** [CharlesWillis3/OctoPrint-MarlinBft#6]
I'm having a problem trying to upload a file (firmware.bin) to my board. The file transfer doesn't seem to progress at all after many minutes of waiting. This is what the transfer log says:

    Starting upload to OctoPrint server
    UPLOAD PROGRESS: 100
    Upload to server done
    Starting transfer to Marlin
    Starting transfer of marlinbft/firmware.bin to firmware.bin on remote
    OK
    pySerial Version: 3.4
    waiting 3000ms after protocol connect
    Connecting: Switching Marlin to binary protocol...
    Connection synced [0], binary protocol version 0.1.0, 96 byte payload buffer
    File Transfer version: 0.1.0, compression: heatshrink

The Marlin LCD says "Receiving file firmware.bin...." but it doesn't progress at all.

![IMG_5779](https://user-images.githubusercontent.com/9397098/89140760-78633680-d4f7-11ea-83ea-762387acbb34.jpeg)

Is there anything I can do to debug the issue?

Thanks

**Comment by mrzottel** [comment]
I had the same issue - in my case (using an SKR 1.4 Turbo) it was a misconfiguration in Marlin. In configuration_adv.h this line was commented out:

#define SDCARD_CONNECTION ONBOARD

(also it was saying LCD instead of ONBOARD).

Accessing the SD using usb worked, but marlin had no access to the SD.

I changed it to the line above, uncommenting it and changing LCD to ONBOARD - and it worked.

**Comment by jpflouret** [comment]
@mrzottel Of course. I didn't think of that. It works now. Thanks for the tip.

As for this issue, I don't think I should close it as it seems incorrect that the plugin just hangs in this case.

**Comment by mrzottel** [comment]
Yes, I found this issue when sending M21 returned "SD card not ok" or something like that. Maybe implement this as a prerequisite check.

**Comment by trippwill** [comment]
@mrzottel thanks for the workaround 👍. @jpflouret  I'm glad it's working for you now 😁.

Adding a check for the SD card is a great suggestion.
